### PR TITLE
Forbid generation of special tokens (excpet EOS)

### PR DESCRIPTION
**Pull request by rfejgin**
We have observed the model sometimes generating special tokens (e.g. a reserved token, audio_context_bos, etc.), which should never happen and could cause artifacts. This was observed in particular when CFG is on.

Here, we explicitly ensure that such tokens are never sampled by forcing the corresponding logits (post-CFG, pre top-k) to `-inf`. `AUDIO_EOS` is permitted since we do need sample it (and generally trim it away before returning the codes to user/codec).

### CODE REVIEW

**Comment by rfejgin** [comment]
I am running some evals to validate the changes are safe but please feel free to review. I'll change to non-draft status when ready.

**Comment by rfejgin** [comment]
Evals look good, I think we can merge.
